adapters/repos/db/vector/hnsw/index.go

```diff
@@ -279,23 +294,32 @@ func New(cfg Config, uc ent.UserConfig,
 		metrics:   NewMetrics(cfg.PrometheusMetrics, cfg.ClassName, cfg.ShardName),
 		shardName: cfg.ShardName,
 
-		randFunc:             rand.Float64,
-		compressActionLock:   &sync.RWMutex{},
-		className:            cfg.ClassName,
-		VectorForIDThunk:     cfg.VectorForIDThunk,
-		TempVectorForIDThunk: cfg.TempVectorForIDThunk,
-		pqConfig:             uc.PQ,
-		bqConfig:             uc.BQ,
-		sqConfig:             uc.SQ,
-		rescoreConcurrency:   2 * runtime.GOMAXPROCS(0), // our default for IO-bound activties
-		shardedNodeLocks:     common.NewDefaultShardedRWLocks(),
+		randFunc:                 rand.Float64,
+		compressActionLock:       &sync.RWMutex{},
+		className:                cfg.ClassName,
+		VectorForIDThunk:         cfg.VectorForIDThunk,
+		MultipleVectorForIDThunk: cfg.MultipleVectorForIDThunk,
+		TempVectorForIDThunk:     cfg.TempVectorForIDThunk,
+		pqConfig:                 uc.PQ,
+		bqConfig:                 uc.BQ,
+		sqConfig:                 uc.SQ,
+		rescoreConcurrency:       2 * runtime.GOMAXPROCS(0), // our default for IO-bound activties
+		shardedNodeLocks:         common.NewDefaultShardedRWLocks(),
 
 		store:                  store,
 		allocChecker:           cfg.AllocChecker,
 		visitedListPoolMaxSize: cfg.VisitedListPoolMaxSize,
+
+		multivector:    atomic.Bool{},
```

Comment:
nitpicking (feel free to ignore): There's no need to initialize zero values in Go, you can simply remove this line and the next